x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -527,6 +530,13 @@ export class TaskRunner<
         category: [this.alertType.producer],
       },
       kibana: {
+        alert: {
+          rule: {
+            execution: {
+              uuid: executionId,
```

Comment:
quick drive-by comment: we'll want this field in all of the alerting events I think: `execute` (done here), but also `execute-start` (may already be done here, as I think we base that off this one), `new-instance`, `recovered-instance`, and `active-instance`.  Guess we'll have to double-check if there are more.

You'll likely find the case where we send out `recovered-instance` events when a rule is disabled, because it can get confusing if we leave them in limbo.  In that case, there is no execution id (happened during disable call, not rule execution), so we should not provide a value there.

Reply:
Thanks for the drive-by @pmuellr! I'm about to pick this back up so appreciate the 👀's. I'll add `uuid` to these other event types, and then get moving on flushing out the test coverage. If there's any specific test cases that you think should be covered let me know and I'll add them, otherwise we can asses coverage come review time.  I'm a little less versed in how specific alert instances behave, so appreciate the heads-up with regards to `recovered-instance`. 🙂

Reply:
We probably will want to have some basic tests that the UUIDs are the same for the events generated from the same execution, and different for events generated from different executions.  I think this would probably just be for the alerting function tests - HMU when you're getting to those bits, I can point out a good place to put them.

Reply:
Added `executionId` to remaining alerting events in a993668.